### PR TITLE
feat(git-utils): getting all amplication commits before creating the branch

### PR DESCRIPTION
**Pull request by tupe12334**
Close: #5143 

## PR Details

This pull request collect all the amplication's commits from the default branch
## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-git-utils/src/git-provider.interface.ts.ts [hunk 1]
```diff
@@ -45,4 +45,5 @@ export interface GitProvider {
   getBranch: (args: GetBranchArgs) => Promise<Branch>;
   createBranch: (args: CreateBranchArgs) => Promise<Branch>;
   getFirstCommitOnBranch: (args: GetBranchArgs) => Promise<Commit>;
+  getMyCommitsList: (args: GetBranchArgs) => Promise<Commit[]>;
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/git-provider.interface.ts.ts:48]:
```suggestion
  getCurrentUserCommitList: (args: GetBranchArgs) => Promise<Commit[]>;
```

File: packages/amplication-git-utils/src/git/github.service.ts [hunk 2]
```diff
@@ -60,6 +62,121 @@ export class GithubService implements GitProvider {
...
+    return {
+      commits: nodes.map(({ oid }) => ({
+        sha: oid,
+      })),
+      hasNextPage,
+      endCursor,
+    };
+  }
+
+  async getAmplicationBotData(): Promise<AmplicationBotData> {
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/git/github.service.ts:164]: this should be internal to the provider

File: packages/amplication-git-utils/src/types.ts [hunk 3]
```diff
@@ -152,3 +152,8 @@ export interface CreateBranchArgs {
 export interface Commit {
   sha: string;
 }
+
+export interface AmplicationBotData {
+  id: string;
+  login: string;
+}
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/types.ts:159]: this should be internal to the provider

File: packages/amplication-git-utils/src/types.ts [hunk 4]
```diff
@@ -152,3 +152,8 @@ export interface CreateBranchArgs {
 export interface Commit {
   sha: string;
 }
+
+export interface AmplicationBotData {
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/types.ts:156]: change the name to something along the lines of GitHubUser
```suggestion
export interface GitHubUser {
```

File: packages/amplication-git-utils/src/git/github.service.ts [hunk 5]
```diff
@@ -60,6 +62,121 @@ export class GithubService implements GitProvider {
...
+          botData,
+          branchName,
+          owner,
+          repositoryName,
+          cursor,
+        });
+      moreCommitsPagination = hasNextPage;
+      commitsList = commitsList.concat(commits); // The list is in ascending order ( newest commits are first )
+      cursor = endCursor;
+    } while (moreCommitsPagination);
```
Comment (yuval-hazaz) [inline packages/amplication-git-utils/src/git/github.service.ts:85]: Should we add another stop here for extreme cases? (e.g. 100 pages?) 
(maybe in the future, we can think of a way to create a snapshot that includes x commits from more than X months ago, so we will not increase the number of commits infinitely)

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
well done! some minor comments

File: packages/amplication-git-utils/src/git/github.service.ts (hunk 5 above)
Comment (tupe12334) [reply packages/amplication-git-utils/src/git/github.service.ts:85]: Its a good idea, but its a separated issue form this

File: packages/amplication-git-utils/src/git/github.service.ts (hunk 2 above)
Comment (tupe12334) [reply packages/amplication-git-utils/src/git/github.service.ts]: I disagree

File: packages/amplication-git-utils/src/types.ts (hunk 3 above)
Comment (tupe12334) [reply packages/amplication-git-utils/src/types.ts]: I disagree

**Pull request merged**